apps/studio/components/interfaces/Database/CronJobs/HttpHeaderFieldsSection.tsx

```diff
@@ -0,0 +1,145 @@
+import clsx from 'clsx'
+import { ChevronDown, Plus, Trash } from 'lucide-react'
+import { useFieldArray } from 'react-hook-form'
+
+import { useParams } from 'common'
+import { FormSectionLabel } from 'components/ui/Forms/FormSection'
+import { useProjectApiQuery } from 'data/config/project-api-query'
+import {
+  Button,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuSeparator,
+  DropdownMenuTrigger,
+  FormControl_Shadcn_,
+  FormField_Shadcn_,
+  FormItem_Shadcn_,
+  FormMessage_Shadcn_,
+  Input_Shadcn_,
+  SheetSection,
+} from 'ui'
+import { CreateCronJobForm } from './CreateCronJobSheet'
+
+interface HTTPHeaderFieldsSectionProps {
+  fieldName: 'values.httpHeaders' | 'values.httpHeaders'
+}
+
+export const HTTPHeaderFieldsSection = ({ fieldName }: HTTPHeaderFieldsSectionProps) => {
+  // gets the fields through form context
+  const { fields, append, remove } = useFieldArray<CreateCronJobForm>({
+    name: fieldName,
+  })
+
+  const { ref } = useParams()
+  const { data: settings } = useProjectApiQuery({ projectRef: ref })
+  const apiService = settings?.autoApiService
+  const apiKey = apiService?.serviceApiKey ?? '[YOUR API KEY]'
+
+  return (
+    <SheetSection>
+      <FormSectionLabel className="lg:!col-span-4">HTTP Headers</FormSectionLabel>
+      <div className="space-y-2">
+        {fields.map((field, index) => (
+          <div key={field.id} className="flex items-center space-x-2">
+            <FormField_Shadcn_
+              name={`${fieldName}.${index}.name`}
+              render={({ field }) => (
+                <FormItem_Shadcn_ className="flex-1">
+                  <FormControl_Shadcn_>
+                    <Input_Shadcn_
+                      {...field}
+                      size="small"
```

Comment:
this header dropdown relevant for the http option?

https://share.cleanshot.com/DLjtQJYf